onboarding/main.go

```diff
@@ -122,3 +158,25 @@ func convertRsaPublicKeyAsPemStr(publicKey *rsa.PublicKey) (string, error) {
 
 	return string(publicKeyPem), nil
 }
+
+func getStorageClusterMetadata(ctx context.Context, operatorNamespace string, clientset *kubernetes.Clientset) metav1.PartialObjectMetadata {
+	var storageClusterMetaDataList metav1.PartialObjectMetadataList
+	storageClusterGVKPath := fmt.Sprintf("/apis/%s/%s/namespaces/%s/%s", "ocs.openshift.io", "v1", operatorNamespace, "storageclusters")
+	instance, err := clientset.RESTClient().Get().AbsPath(storageClusterGVKPath).DoRaw(ctx)
+	if err != nil {
+		klog.Errorf("failed to get storage cluster metadata: %v", err.Error())
+		os.Exit(1)
```

Comment:
Please do not use `os.Exit` inside nested functions. 
Return an error to the caller and act on it on the call site (upper-level main).